docs/Configuration.md

```diff
@@ -1183,6 +1183,74 @@ If you also have specified [`rootDir`](#rootdir-string), the resolution of this
 
 :::
 
+### `plugins` \[array&lt;Plugin&gt;]
```

Comment:
Can we somehow combine your implementation with existing https://jestjs.io/docs/watch-plugins to have a single Plugin API where user can define:
- existing watch hooks: `apply`, `getUsageInfo`, `run`
- `transform` hook
- `config` hook
- `configResolved` hook
- `configureJest` hook

Then we can add deprecation annotation and message to watch plugins and add documentation to instruct users to move to new Plugin API.